pygmt/tests/test_figure.py

```diff
@@ -251,3 +251,12 @@ def test_figure_set_display_invalid():
     """
     with pytest.raises(GMTInvalidInput):
         set_display(method="invalid")
+
+
+def test_figure_icc_gray():
+    """
+    Check if icc_gray parameter works correctly if used.
+    """
+    fig = Figure()
+    fig.basemap(region=[0, 1, 0, 1], projection="X1c/1c", frame=True)
+    fig.psconvert(icc_gray=True, prefix="Test")
```

Comment:
Also need to add `import pytest` at L14 for this to work.
```suggestion
    with pytest.warns(expected_warning=FutureWarning) as record:
        fig.psconvert(icc_gray=True, prefix="Test")
        assert len(record) == 1  # check that only one warning was raised
```